### ISSUE TITLE
Intentionally left blank pages - even number of pages for each chapter

### ISSUE THREAD

**Issue by burkaysucu** [QuestPDF/QuestPDF#675]
First of all, thank you for all the effort you put into this great library.

Our data consist of chapters. For each chapter, we use a `page` container to handle the different settings. We want each chapter end with an even-numbered page for printing purposes.

We tried using a dynamic component and called `pageBreak` in it. However, `pageBreak` creates a new page which causes an exception for the dynamic component. Other than the dynamic component, we could not find a way to get the page count of a chapter.

Is there a way to achieve this?

**Comment by MarcinZiabek** [comment]
Hi 😀

Instead of generating `PageBreak` in the `Dynamic` component, please return content that takes entire space, e.g. with the `Extend` element. 

I haven't tested this suggestion. If it happens to now work, I will try to device another workaround.

**Comment by burkaysucu** [comment]
Hello :) Thank you for the fast response. 

We tried the following piece of code with no success:

    ...
    column.Item().Element(Chapter.Compose);
    column.Item().Extend().Background(Colors.Red.Accent2); // this extends the container on the last page
    column.Item().Dynamic(new EmptyEvenLastPage());


and the Compose method of `EmptyEvenLastPage` class:

    public DynamicComponentComposeResult Compose(DynamicContext context)
    { 
        // context.AvailableSize has zero height at this point
        var content = context.CreateElement(container =>
        {
            if (context.TotalPages %2 == 1)
            {
                container.Background(Colors.Red.Accent2).Text("Intentionally left blank"); 
            }
        });

        return new DynamicComponentComposeResult
        {
            Content = content,
            HasMoreContent = false
        };
    }

**Comment by burkaysucu** [comment]
We tried a ton of other things. Still no luck. Do you have any other suggestion/workaround other them double compiling?

**Comment by MarcinZiabek** [comment]
@burkaysucu Please test the solution presented in the https://github.com/QuestPDF/QuestPDF/issues/706 issue. It is most likely very close.

Also, I am open to discussing introducing a native API for this feature. Do you have any suggestions?

**Comment by MarcinZiabek** [comment]
Not exactly, as far as I understand, the behavior was implemented successfully. Insteady of the `PrageBreak` API, the author has used the `ExtendVertical` element.

**Comment by burkaysucu** [comment]
Now I read it more carefully and realized that it is a slightly different request. I am sorry for missing it before.

I will try it and let you know. Thank you again:)

**Comment by MarcinZiabek** [comment]
No worries!

I would like to brainstorm a little bit about potential native solutions. I am always careful when introducing new APIs. Everything related to conditional content rendering has the potential of completely breaking the layout algorithm or producing unstable states.
